python/ray/serve/tests/test_api.py

```diff
@@ -53,6 +67,157 @@ async def __call__(self):
+
+        serve.run(Model.bind())
+        resp = requests.get("http://127.0.0.1:8000/").json()
+        expected_log_regex = [".*model_info_level.*"]
+        check_log_file(resp["log_file"], expected_log_regex)
+
+        # Make sure 'model_debug_level' log content does not exist
+        with pytest.raises(AssertionError):
+            check_log_file(resp["log_file"], [".*model_debug_level.*"])
+
+        serve.run(Model.options(logging_config={"log_level": "DEBUG"}).bind())
+        resp = requests.get("http://127.0.0.1:8000/").json()
+        expected_log_regex = [".*model_info_level.*", ".*model_debug_level.*"]
+        check_log_file(resp["log_file"], expected_log_regex)
+
+    def test_logs_dir(self, serve_instance):
+
+        logger = logging.getLogger("ray.serve")
+
+        @serve.deployment
+        class Model:
+            def __call__(self, req: starlette.requests.Request):
+                logger.info("model_info_level")
+                return {
+                    "logs_path": logger.handlers[1].baseFilename,
+                }
+
+        serve.run(Model.bind())
+        resp = requests.get("http://127.0.0.1:8000/").json()
+
+        paths = resp["logs_path"].split("/")
+        paths[-1] = "new_dir"
+        new_log_dir = "/".join(paths)
+
+        serve.run(Model.options(logging_config={"logs_dir": new_log_dir}).bind())
+        resp = requests.get("http://127.0.0.1:8000/").json()
+        assert "new_dir" in resp["logs_path"]
+
+        check_log_file(resp["logs_path"], [".*model_info_level.*"])
+
+    @pytest.mark.parametrize(
+        "access_type", ["ALL", "FILE_ONLY", "STREAM_ONLY", "DISABLE"]
+    )
+    def test_access_log(self, serve_instance, access_type):
+        logger = logging.getLogger("ray.serve")
+
+        @serve.deployment(logging_config={"access_log": access_type})
+        class Model:
+            def __call__(self, req: starlette.requests.Request):
+                if access_type == "ALL":
+                    assert len(logger.handlers) == 2
+                    assert isinstance(logger.handlers[0], logging.StreamHandler)
+                    assert isinstance(
+                        logger.handlers[1], logging.handlers.RotatingFileHandler
+                    )
+                elif access_type == "FILE_ONLY":
+                    assert len(logger.handlers) == 1
+                    assert isinstance(
+                        logger.handlers[0], logging.handlers.RotatingFileHandler
+                    )
+                elif access_type == "STREAM_ONLY":
+                    assert len(logger.handlers) == 1
+                    assert isinstance(logger.handlers[0], logging.StreamHandler)
+                else:
+                    assert len(logger.handlers) == 0
+                return
+
+        serve.run(Model.bind())
+
+        resp = requests.get("http://127.0.0.1:8000/")
+        assert resp.status_code == 200
+
+    def test_application_logging_overwrite(self, serve_instance):
```

Comment:
please also add a test for overwriting per-deployment

Reply:
done!